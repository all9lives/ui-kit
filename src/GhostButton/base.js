import React, { Component } from 'react'
import Unit from '../utils/Unit'
import Color from '../utils/Color'
import Size from '../utils/Size'
import View from '../View'
import Text from '../Text'

export const StyledView = View.extend`
  border-style: solid;
  border-radius: ${Unit.convert(Size.title)};
  border-width: 1px;
  border-color: ${props => Color[props.color] || props.color};
  padding: ${`${Unit.convert(Size.spacing * 1.5)} ${Unit.convert(Size.spacing * 5)}`};
`

export const StyledText = Text.extend`
  color: ${props => Color[props.color] || props.color};
  background-color: transparent;
`

export class Base extends Component {
  render () {
    const { title, ...props } = this.props
    return (
      <StyledView {...props}>
        <StyledText {...props}>
          {title}
        </StyledText>
      </StyledView>
    )
  }
}
