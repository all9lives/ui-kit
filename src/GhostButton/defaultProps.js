import Color from '../utils/Color'

export default {
  title: 'Button',
  color: Color.black
}
