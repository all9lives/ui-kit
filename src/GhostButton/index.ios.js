import GhostButton from './index.android'

export default GhostButton
