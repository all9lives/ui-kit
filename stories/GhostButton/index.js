import React from 'react'
import { action } from '@storybook/addon-actions'
import createStory from '../../storybook/createStory'
import GhostButton from '../../src/GhostButton'

createStory('GhostButton', module)
  .add('with Apple', () => <GhostButton title='Apple' />)
  .add('with onPress', () => <GhostButton onPress={action('Pressed')} />)
  .add('with color `white`', () => <GhostButton color='white' />)
