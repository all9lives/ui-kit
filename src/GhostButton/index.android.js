import React from 'react'
import { TouchableOpacity } from 'react-native'
import { Base } from './base'
import defaultProps from './defaultProps'

const GhostButton = ({ onPress, ...props }) => (
  <TouchableOpacity onPress={onPress}>
    <Base {...props} />
  </TouchableOpacity>
)
GhostButton.defaultProps = defaultProps

export default GhostButton
