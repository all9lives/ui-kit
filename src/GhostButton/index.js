import React from 'react'
import { Base } from './base'
import defaultProps from './defaultProps'

const GhostButton = ({ onClick, onPress, ...props }) => (
  <span onClick={onClick || onPress}>
    <Base {...props} />
  </span>
)

GhostButton.defaultProps = defaultProps

export default GhostButton
