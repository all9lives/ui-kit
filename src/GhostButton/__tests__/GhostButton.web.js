/* eslint-env jest */
import 'jest-styled-components'
import React from 'react'
import { shallow } from 'enzyme'
import GhostButton from '../index.js'

describe('GhostButton.web', () => {
  it('renders a snapshot', () => {
    const wrapper = shallow(
      <GhostButton>Apple</GhostButton>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
