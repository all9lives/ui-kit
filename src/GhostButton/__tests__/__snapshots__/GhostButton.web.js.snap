// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GhostButton.web renders a snapshot 1`] = `
<span>
  <Base
    color="#191919"
    title="Button"
  >
    Apple
  </Base>
</span>
`;

exports[`GhostButton.web renders a snapshot with \`color:white\` 1`] = `
<span>
  <Base
    color="white"
    title="Button"
  >
    Apple
  </Base>
</span>
`;

exports[`GhostButton.web renders a snapshot with \`onPress\` 1`] = `
<span
  onClick={[Function]}
>
  <Base
    color="#191919"
    title="Button"
  >
    Apple
  </Base>
</span>
`;
